Sources/Layout/MessagesCollectionViewFlowLayout+MessageContainerView.swift

```diff
@@ -119,38 +119,18 @@ extension MessagesCollectionViewFlowLayout {
 extension MessagesCollectionViewFlowLayout {
 
     internal func _messageLabelInsets(for message: MessageType, at indexPath: IndexPath) -> UIEdgeInsets {
-        guard let cachedInsets = currentLayoutContext.messageLabelInsets else {
-            let insets = messageLabelInsets(for: message, at: indexPath)
-            currentLayoutContext.messageLabelInsets = insets
-            return insets
-        }
-        return cachedInsets
+        return messageLabelInsets(for: message, at: indexPath)
     }
 
     internal func _messageContainerPadding(for message: MessageType, at indexPath: IndexPath) -> UIEdgeInsets {
-        guard let cachedPadding = currentLayoutContext.messageContainerPadding else {
-            let padding = messageContainerPadding(for: message, at: indexPath)
-            currentLayoutContext.messageContainerPadding = padding
-            return padding
-        }
-        return cachedPadding
+        return messageContainerPadding(for: message, at: indexPath)
     }
 
     internal func _messageContainerMaxWidth(for message: MessageType, at indexPath: IndexPath) -> CGFloat {
-        guard let cachedMaxWidth = currentLayoutContext.messageContainerMaxWidth else {
```

Comment:
We had all these internal methods to not repeat calculations. We may need `messageContainerMaxWidth` in `messageContainerSize`, `cellTopLabelSize`, `cellBottomLabelSize`, so we would need to call this 3 times

Reply:
That's why I put the `currentLayoutContext` at the global scope to share the values

Reply:
`currentLayoutContext` only assign in `newCellLayoutContext(for message: MessageType, at indexPath: IndexPath)`, I don't know wether would leads to chaos when layout. I can change these internal method that only calculate when not in cache.

Reply:
Ugh, that is a good point 👍 I remember thinking about this. I can't be sure they will always be available in the property. I would need to know more about the order of `UICollectionView` method calls.  Better to be safe...

The cache will be empty because we only store it after completing all the calculations

Reply:
@SD10 , ping 😄 .